File: 02._Data_Files/01a Assignment/Python/main.py
# %%
import os
from pathlib import Path
from file_type_reader import read_JSON_from_file, read_CSV_from_file_pandas, read_XML_from_file, read_YAML_from_file
import xml.etree.ElementTree as ET
from IPython.display import display, HTML

# %%
json_file_path = os.path.join(Path(__file__).parent.parent.parent, 'me.json')
csv_file_path = os.path.join(Path(__file__).parent.parent.parent, 'me.csv')
xml_file_path = os.path.join(Path(__file__).parent.parent.parent, 'me.xml')
yaml_file_path = os.path.join(Path(__file__).parent.parent.parent, 'me.yaml')
txt_file_path = os.path.join(Path(__file__).parent.parent.parent, 'me.txt')

json_data = read_JSON_from_file(json_file_path)
csv_data = read_CSV_from_file_pandas(csv_file_path)
xml_data = read_XML_from_file(xml_file_path)
yaml_data = read_YAML_from_file(yaml_file_path)
txt_data = read_YAML_from_file(txt_file_path)

if xml_data is not None:
    # Display the XML element
    display(HTML(ET.tostring(xml_data, encoding='utf-8').decode()))

if yaml_data is not None:
    print(yaml_data)

# %%

File: 02._Data_Files/01a Assignment/JS/main.js
// Import the functions
const { decode } = require('punycode');
const readerFunctions = require('./fileReader.js');
const path = require('path');

const dataFolder = path.join(__dirname, '..', '..');

// Subfolders for each data type
const yamlFolderPath = path.join(dataFolder, 'me.yaml');
const jsonFolderPath = path.join(dataFolder, 'me.json');
const xmlFolderPath = path.join(dataFolder, 'me.xml');
const csvFolderPath = path.join(dataFolder, 'me.csv');
const txtFolderPath = path.join(dataFolder, 'me.txt');

console.log(dataFolder)

// Example usage
const yamlData = readerFunctions.readYAMLFromFile(yamlFolderPath);
console.log(yamlData);

const jsonData = readerFunctions.readJSONFromFile(jsonFolderPath);
console.log(jsonData);

const xmlData = readerFunctions.readXMLFromFile(xmlFolderPath);
console.log(xmlData);

const csvData = readerFunctions.readCSVFromFile(csvFolderPath);
console.log(csvData);

const txtData = readerFunctions.readTextFile(txtFolderPath);
console.log(txtData);

